src/components/study-activity-calendar/study-activity-calendar.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from 'react';
+import { CalendarGraph, DailyScore } from '@/components/calendar-graph/calendar-graph';
+import { useStudyActivity } from '@/hooks/api/use-study-activity';
+import './study-activity-calendar.scss';
+
+// year -> DailyScore[]
+type AnnualDailyScores = Record<number, DailyScore[]>;
+
+interface StudyActivityCalendarProps {
+  className?: string;
+}
+
+export const StudyActivityCalendar = ({ className = '' }: StudyActivityCalendarProps) => {
+  const [currentYear, setCurrentYear] = useState(new Date().getFullYear());
+  const [annualDailyScores, setAnnualDailyScores] = useState<AnnualDailyScores>({});
+  const { getAnnualDailyStudyRecords } = useStudyActivity();
+
+  // fetch annual daily scores, first on load then on current year change
+  useEffect(() => {
+    fetchAnnualStudyRecordsAndRefresh(currentYear);
+  }, [currentYear, setAnnualDailyScores]);
```

Comment:
is `setAnnualDailyScores` in dep needed?

Reply:
Lol, nope. It's fixed now